tests/rptest/infinite_retention/infinite_retention_test.py

```diff
@@ -0,0 +1,794 @@
+            3)
+        self.logger.info(f"Manifests uploads: {stats['manifest_uploads']}, "
+                         f"Expected not less than: {expect_manifest_uploads}")
+        chk.add_arg("produce_duration", int(produce_duration))
+        chk.add_arg("manifest_upload_interval",
+                    self.params.manifest_upload_interval)
+        chk.add_check(
+            "For spillover active tests manifest uploads should be "
+            "significant",
+            "manifest_uploads > produce_duration // manifest_upload_interval + 3",
+        )
+        # Do all required checks with Redpanda active
+        chk.conduct_checks()
+        # Run rp_storage_tool
+        self.redpanda.stop_and_scrub_object_storage()
+        # Validate checks and generate AssertionError if needed
+        if chk.assert_results():
+            self.logger.info(chk.get_summary_as_text())
+
+    @cluster(num_nodes=6)
+    def long_retention_with_spillover_test(self):
+        iteration_count = self.params.total_iterations
+        iteration = 1
+
+        # Do additional configuration
+        self.rpk.alter_topic_config(self.topic, 'retention.local.target.bytes',
+                                    self.params.local_retention_bytes)
+
+        # Log important parameters
+        self._log_parameters()
+
+        # Create consumer with no limit
+        # consumer = self._create_group_consumer()
+        # SeqConsumer (broken)
+        consumer = self._create_seq_consumer()
+        # # Start it in a clean node
+        # consumer.start()
+
+        # Producer
+        producer = self._create_producer()
+
+        # Prepare to save timigs
+        iteration_timings = {
+            "high_watermark": {},
+            "produce": {},
+            "manifest_uploads": {},
+            "segment_uploads": {},
+            "sp_mn_uploads_total": {},
+            "consumed_messages": {},
+            "consume": {},
+            "checks": {},
+            "scrub": {},
+            "restart": {}
+        }
+
+        # Announce scrub_exception
+        scrub_exception = None
+        # Do iterations of producing messages
+        while iteration <= iteration_count:
+            self.logger.info(f"Starting iteration {iteration},"
+                             f"runtime {self.params.target_runtime}s")
+            # Consumer are to be created each time and released at the end
+
+            # rand_consumer = self._create_rnd_consumer()
+            # consumer = self._create_seq_consumer()
+            # consumer = self._create_group_consumer()
+
+            # Producer
+            produce_start = time.time()
+            producer.start()
+            # wait for messages to appear in queue
+            producer.wait_for_acks(10000, timeout_sec=60, backoff_sec=10)
+            producer.wait_for_offset_map()
+            # wait and stop
+            # producer.wait(timeout_sec=self.params.expected_producer_duration)
```

Comment:
commented out lines here, but also `producer.stop()` calls below?  It's not obvious which lines are intentionally commented out as future work, and which might have been left by accident.

Reply:
This is a part of new approach that produces and consumes data at the same time. The point is that there should be more consumers than producers.